### PR TITLE
[BH-1899] Focus timer basic

### PR DESCRIPTION
**Pull request by ghost**
Added a basic version of the Focuse timer function. This basic version allows the user to run the focus timer with only the default parameters. Focus time is set at 25 minutes, short break at 5 minutes for 10 focus intervals.

**Your checklist for this pull request**

Make sure that this PR:
- [ ] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

File: products/BellHybrid/apps/application-bell-focus-timer/data/FocusContract.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+      public:
+        virtual ~View() noexcept = default;
+    };
+
+    class SettingsPresenter : public BasePresenter<View>
+    {
+      public:
+        virtual ~SettingsPresenter() noexcept                                           = default;
+        virtual auto getPagesProvider() const -> std::shared_ptr<gui::ListItemProvider> = 0;
```
Comment (dawidwojtas-mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/data/FocusContract.hpp:23]: Consider using `auto fun() -> type` for all declarations or 
```suggestion
        virtual std::shared_ptr<gui::ListItemProvider> getPagesProvider() const  = 0;
```
to be consistent in this file.

File: products/BellHybrid/apps/application-bell-focus-timer/data/FocusTimerStyle.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#include <Style.hpp>
+#include "widgets/BellBaseLayout.hpp"
+
+namespace app::focusTimerStyle
+{
+    namespace runningStyle
+    {
+        namespace progress
+        {
+            constexpr inline auto radius                   = 192;
```
Comment (dawidwojtas-mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/data/FocusTimerStyle.hpp:15]: To be consistent here use `inline constexpr auto` here for all variables. Also, add `U` at the end of numbers.
```suggestion
            inline constexpr auto radius                   = 192U;
```

File: products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.hpp [hunk 3]
```diff
@@ -17,10 +22,34 @@ namespace app::focus
...
+        void onFocusSessionStarted() override;
+        void onFocusSessionFinished() override;
+        void onShortBreakStarted() override;
+        void onShortBreakFinished() override;
+        void pause() override;
+        void resume() override;
 
       private:
         std::unique_ptr<FocusTimerContract::Presenter> presenter;
+        gui::VBox *mainVBox                   = nullptr;
```
Comment (dawidwojtas-mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.hpp:38]: Here I'd use
```suggestion
        gui::VBox *mainVBox{nullptr};
```
but it is up to you ;)

**Review by dawidwojtas-mudita — APPROVED** [review]

File: products/BellHybrid/apps/application-bell-focus-timer/data/FocusCommon.hpp [hunk 4]
```diff
@@ -4,15 +4,26 @@
...
 
 #include <AppWindowConstants.hpp>
+#include <Paths.hpp>
+#include <filesystem>
+#include <string>
 
 namespace app::focus
 {
     inline constexpr auto focusDBRecordName = "FocusTimer";
+    constexpr std::chrono::minutes emptyInterval{0};
```
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/data/FocusCommon.hpp:14]: is this used anywhere?

File: products/BellHybrid/apps/application-bell-focus-timer/data/FocusContract.hpp [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        virtual bool isTimerStopped()                                           = 0;
+        virtual void start()                                                    = 0;
+        virtual void stop()                                                     = 0;
+        virtual void pause()                                                    = 0;
+        virtual void resume()                                                   = 0;
+        virtual void abandon()                                                  = 0;
+        virtual void finish()                                                   = 0;
+        virtual void onBeforeShow()                                             = 0;
+    };
+} // namespace app::focus::contract
```
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/data/FocusContract.hpp:46]: is this file needed? I see there are separate contracts for dedicated presenters.

File: products/BellHybrid/apps/application-bell-focus-timer/presenter/FocusTimerPresenter.hpp [hunk 6]
```diff
@@ -16,28 +25,84 @@ namespace app::focus
...
+        std::unique_ptr<app::TimerWithCallbacks> timer;
+        sys::TimerHandle betweenSessionTimer;
+        std::unique_ptr<AbstractTimeModel> timeModel;
+        std::chrono::minutes focusSessionDuration;
+        std::chrono::minutes shortBreakDuration;
+        std::uint32_t allFocusSessionsCount;
+        std::uint32_t focusSessionsLeft;
+        FocusTimerPhase currentTimerPhase = FocusTimerPhase::FocusTime;
+
+        static constexpr std::chrono::milliseconds betweenSessionDelayTime{5000};
```
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/presenter/FocusTimerPresenter.hpp:103]: please move this to an anonymous namespace in the .cpp file

File: products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.cpp [hunk 7]
```diff
@@ -2,10 +2,27 @@
...
+namespace
+{
+    constexpr auto focusTimerProgressTimerName{"FocusProgressTimer"};
+    constexpr std::chrono::seconds focusTimerProgressTimerPeriod{1};
+    constexpr auto focusTimerProgressMode{app::ProgressCountdownMode::Increasing};
+    constexpr auto focusTimerText{"Focus timer"};
+    constexpr auto timeToFocusText{"Time to focus"};
+    constexpr auto shortBreakTimeText{"Short break"};
+    constexpr auto timeForBreakText{"Time for a break"};
+    constexpr auto endOfAllSessionText{"Well done!"};
```
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.cpp:23]: I saw that there are already translations but I can add them with my changes :)

**Review by Maciej-Mudita — APPROVED** [review]
